### PR TITLE
Bugfix/bphh 1057/add specs and delete pipeline

### PR DESCRIPTION
**Pull request by sunto**
## Description

Refactored the code for a more succinct upload process.  No syncing of submission_data payload with supporting documents to reduce edge cases.
More consistent removal of item (even if save is not triggered or svae is triggered).
Updated the parsing of signatures information.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

(https://hous-bssb.atlassian.net/browse/BPHH-992)
(https://hous-bssb.atlassian.net/browse/BPHH-1057)

## Steps to QA
file uploads should still work.
computed compliance should still work.
After uploading, and compliance is run, if you remove the document the compliance will clear.
If you delete persisted file and do not save, you can refresh the screen and download it.

### CODE REVIEW

File: app/frontend/components/shared/chefs/additional-formio/providers/storage/s3custom.js [hunk 1]
```diff
@@ -25,48 +25,72 @@ const s3custom = function Provider(formio) {
...
           },
         }
+
+        const persistedResponse = await persistFileUpload(
+          options?.config?.formCustomOptions?.persistFileUploadAction,
+          options?.config?.formCustomOptions?.persistFileUploadUrl,
+          presignedPermitApplicationPayload,
+          presignedPayload
+        )
+        console.log("***", persistedResponse)
```
Comment (LaurelOlson) [inline app/frontend/components/shared/chefs/additional-formio/providers/storage/s3custom.js:66]: cruft

File: app/controllers/api/storage_controller.rb [hunk 2]
```diff
@@ -24,32 +24,14 @@ def download
   end
 
   def delete
-    #TODO: change if the item is a cache or storage client
-    #if the object is already persisted to storage, we remove it off supporting documents itself
     if params[:id].start_with?("cache/")
       # if we use files instead of simple files, we'd need to send back a presigned url directly
```
Comment (LaurelOlson) [inline app/controllers/api/storage_controller.rb:28]: is this comment still relevant?

File: app/frontend/components/shared/permit-applications/requirement-form.tsx [hunk 3]
```diff
@@ -225,6 +225,17 @@ export const RequirementForm = observer(
       setFirstComponentKey(firstComponent.key)
     }
 
+    let permitAppOptions = {
+      ...defaultOptions,
+      ...(isDraft ? {} : { readOnly: true }),
+    }
+    permitAppOptions.componentOptions.simplefile.config["formCustomOptions"] = {
+      // documentParent: "PermitApplication",
+      // documentParentId: permitApplication.id,
```
Comment (LaurelOlson) [inline app/frontend/components/shared/permit-applications/requirement-form.tsx:234]: cruft

File: app/frontend/utils/formio-component-traversal.ts [hunk 4]
```diff
@@ -56,6 +56,17 @@ export const combineComplianceHints = (
   formattedComplianceData
 ) => {
   let updatedJson = formJson
+  //special step - utilize the fileKey variable in simplefile to pass the file through, this is done this way to not modify the underlying chefs simplefile implementation
+  //currently better than doing on on the requirement_json service as we will likley modify items in the long run
+  const fileComponents = findFileComponents(updatedJson.components)
+  fileComponents.forEach((fileComponent) => {
+    if (fileComponent.multiple) {
+      //For multiple files do something else
```
Comment (LaurelOlson) [inline app/frontend/utils/formio-component-traversal.ts:64]: does this need a `TODO`?

File: app/models/concerns/form_supporting_documents.rb [hunk 5]
```diff
@@ -2,6 +2,46 @@ module FormSupportingDocuments
...
+  included do
+    has_many :supporting_documents, dependent: :destroy
+    has_many :active_supporting_documents,
+             ->(permit_application) do
+               where(
+                 id:
+                   permit_application.find_file_fields_and_transform!(
+                     permit_application.submission_data,
+                     [],
+                   ) { |file_field_key, file_array| file_array.map { |fa| fa["model_id"] } },
```
Comment (LaurelOlson) [inline app/models/concerns/form_supporting_documents.rb:14]: could extract this to a method

**Review by LaurelOlson — COMMENTED** [review]
Some UI fixes:

- the `x` button for deleting the file should use the 'pointer' cursor
- once the file is deleted, the compliance message disappears so the popover is empty (it should revert to the default message - "This field has Auto-Compliance capability")
Screen Shot 2024-04-04 at 10 52 50 AM

File: app/blueprints/supporting_document_blueprint.rb [hunk 6]
```diff
@@ -1,3 +1,18 @@
 class SupportingDocumentBlueprint < Blueprinter::Base
   fields :file_name, :file_url, :file_size
+
+  view :form_io_details do
+    #update later since this particular call will not utilize humps on front end
```
[the remark this answers is not in the document]
Comment (sunto) [reply app/blueprints/supporting_document_blueprint.rb:5]: yes, I'll put that onto the modelId itself

**Pull request merged**